File: 00606. Construct String from Binary Tree/00606.java
class Solution {
    public String tree2str(TreeNode root) {
        if (root == null) {
            return "";
        } else {
            String ret = root.val + "";
            
            if (root.left != null || root.right != null) {
                ret += "(" + tree2str(root.left) + ")";

                if (root.right != null) {
                    ret += "(" + tree2str(root.right) + ")";
                }
            }
            
            return ret;
        }
    }
}

File: 00606. Construct String from Binary Tree/00606.go
// https://leetcode.com/problems/construct-string-from-binary-tree

func tree2str(root *TreeNode) string {
    if root == nil {
        return ""
    }

    s := strconv.Itoa(root.Val) 
    if root.Right != nil {
        s += "(" + tree2str(root.Left) + ")(" + tree2str(root.Right) + ")"
    } else if root.Left != nil {
        s += "(" + tree2str(root.Left) + ")"
    }
    return s
}
